webui/react/src/stores/experiments.tsx

```diff
@@ -0,0 +1,67 @@
+import { Map } from 'immutable';
+import React, { createContext, PropsWithChildren, useCallback, useContext, useState } from 'react';
+
+import { getExperiments } from 'services/api';
+import { V1Pagination } from 'services/api-ts-sdk';
+import { GetExperimentsParams } from 'services/types';
+import { ExperimentItem } from 'types';
+import handleError from 'utils/error';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+type ExperimentPagination = {
+  experiments: ExperimentItem[];
+  pagination?: V1Pagination;
+};
+
+type ExperimentsContext = {
+  experimentsIndex: Map<string, ExperimentPagination>;
+  updateExperimentsIndex: (ws: Map<string, ExperimentPagination>) => void;
+};
+
+const ExperimentsContext = createContext<ExperimentsContext | null>(null);
+
+const encodeArgs = (args: { [key: string]: any }): string => {
+  const orderedKeys = Object.keys(args).sort();
+  return orderedKeys.map((key: string) => `${key}=${JSON.stringify(args[key])}`).join(',');
+};
+
+export const ExperimentsProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const [experimentsIndex, updateExperimentsIndex] = useState<Map<string, ExperimentPagination>>(
+    Map<string, ExperimentPagination>(),
+  );
+  return (
+    <ExperimentsContext.Provider value={{ experimentsIndex, updateExperimentsIndex }}>
+      {children}
+    </ExperimentsContext.Provider>
+  );
+};
+
+export const useFetchExperiments = (
+  filters: GetExperimentsParams,
```

Comment:
since this also includes a `limit` flag which can also operate as a flag, maybe change `filters` to args for consistency with above? or change to `params` and `encodeParams`?

Reply:
I like `params` and `encodeParams`, will do